src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -1232,11 +1234,12 @@ void HierRTLMP::runHierarchicalMacroPlacement(Cluster* parent)
     }
     // for other clusters
     soft_macro_id_map[cluster->getName()] = macros.size();
-    SoftMacro* soft_macro = new SoftMacro(cluster);
+    std::unique_ptr<SoftMacro> soft_macro
+        = std::make_unique<SoftMacro>(cluster);
```

Comment:
this is a good place for auto to avoid repeating the same info twice
```
auto soft_macro = std::make_unique<SoftMacro>(cluster);
```
Similar elsewhere